com.ibm.wala.core.tests/src/com/ibm/wala/core/tests/cha/ExclusionsTest.java

```diff
@@ -34,4 +44,111 @@ public void testExclusions() throws IOException {
+     TypeReference buttonRef = TypeReference.findOrCreate(ClassLoaderReference.Application,
+        StringStuff.deployment2CanonicalTypeString("java.awt.Button"));
+    ExclusionSpecification exclusions = new ExclusionSpecification(ExclusionSpecification.Kind.INCL_ONLY);
+    Collection<String> inclusions1 = new HashSet<String>();
+    inclusions1.add("java/awt/Button");
+    exclusions.addInclusionSet(new ExplicitSetOfClasses(inclusions1));
+    Collection<String> inclusions2 = new HashSet<String>();
+    inclusions2.add("java/awt/Canvas");
+    exclusions.addInclusionSet(new ExplicitSetOfClasses(inclusions2));
+    assertFalse(exclusions.contains(buttonRef.getName().toString().substring(1)));
+    }
+
+  @Test
+  public void testMultipleInclusionsSingleExclusions() throws IOException {
+    TypeReference buttonRef = TypeReference.findOrCreate(ClassLoaderReference.Application,
+        StringStuff.deployment2CanonicalTypeString("java.awt.Button"));
+    ExclusionSpecification exclusions =
+        new ExclusionSpecification(ExclusionSpecification.Kind.INCL_OVERRIDE_EXCL);
+    InputStream exclusionStream =
+        new FileInputStream((new FileProvider()).getFile("GUIExclusions.txt"));
+    exclusions.addExclusionSet(new FileOfClasses(exclusionStream));
+    Collection<String> inclusions1 = new HashSet<String>();
+    inclusions1.add("java/awt/Button");
+    exclusions.addInclusionSet(new ExplicitSetOfClasses(inclusions1));
+    Collection<String> inclusions2 = new HashSet<String>();
+    inclusions2.add("java/awt/Canvas");
+    exclusions.addInclusionSet(new ExplicitSetOfClasses(inclusions2));
+    assertFalse(exclusions.contains(buttonRef.getName().toString().substring(1)));
+    }
+
+  @Test
+  public void testMultipleInclusionsMultipleExclusions() throws IOException {
+    TypeReference buttonRef = TypeReference.findOrCreate(ClassLoaderReference.Application,
+        StringStuff.deployment2CanonicalTypeString("java.awt.Button"));
+    ExclusionSpecification exclusions =
+        new ExclusionSpecification(ExclusionSpecification.Kind.INCL_OVERRIDE_EXCL);
+    InputStream exclusionStream1 =
+        new FileInputStream((new FileProvider()).getFile("GUIExclusions.txt"));
+    exclusions.addExclusionSet(new FileOfClasses(exclusionStream1));
+    InputStream exclusionStream2 =
+        new FileInputStream((new FileProvider()).getFile("GUICorbaExclusions.txt"));
+    exclusions.addExclusionSet(new FileOfClasses(exclusionStream2));
+    Collection<String> inclusions1 = new HashSet<String>();
+        inclusions1.add("java/awt/Button");
+    exclusions.addInclusionSet(new ExplicitSetOfClasses(inclusions1));
+    Collection<String> inclusions2 = new HashSet<String>();
+    inclusions2.add("java/awt/Canvas");
+    exclusions.addInclusionSet(new ExplicitSetOfClasses(inclusions2));
+    assertFalse(exclusions.contains(buttonRef.getName().toString().substring(1)));
+    }
+  
+  @Test(expected = IllegalArgumentException.class)
```

Comment:
That might make sense; there are several ways to achieve this, which are perhaps better discussed in person (I started typing in a comment describing the possible options and it got rather lengthy...)